snuba/migrations/snuba_migrations/metrics/0001_metrics_sets.py

```diff
@@ -0,0 +1,160 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import (
+    AggregateFunction,
+    Array,
+    Column,
+    DateTime,
+    Nested,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.storages.tags_hash_map import INT_TAGS_HASH_MAP_COLUMN
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+sets_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("metric_id", UInt(64)),
+    Column("granularity", UInt(32)),
+    Column("timestamp", DateTime()),
+    Column("retention_days", UInt(16)),
+    Column("value", AggregateFunction("uniqCombined64", [UInt(64)])),
+]
+
+sets_mv_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("metric_id", UInt(64)),
+    Column("granularity", UInt(32)),
+    Column("timestamp", DateTime()),
+    Column("retention_days", UInt(16)),
+    Column(
+        "tags",
+        Nested([Column("key", Array(UInt(64))), Column("value", Array(UInt(64)))]),
+    ),
+    Column("value", AggregateFunction("uniqCombined64", [UInt(64)])),
+]
+
+
+matview_statement = """
+SELECT
+    org_id,
+    project_id,
+    metric_id,
+    60 as granularity,
```

Comment:
We can have multiple materialization with different granularity in the same table by changing the group by.

Having multiple materializations in the same table allows us to merge aggregation states at different granularities in the same query: query from 3 pm of monday to 3 pm of wednesday can be done by composing hourly migrations between 3 pm monday to midnight then daily aggregation for tuesday and then hourly aggregations from midnight to 3 pm wednesday.